Userland/Applications/Assistant/Providers.cpp

```diff
@@ -144,7 +145,20 @@ void FileProvider::query(DeprecatedString const& query, Function<void(Vector<Non
                 if (match_result.score < 0)
                     continue;
 
-                results.append(adopt_ref(*new FileResult(path, match_result.score)));
+                if (sorted_results.size() < MAX_SEARCH_RESULTS || match_result.score > sorted_results.peek_min_key()) {
+                    if (sorted_results.size() == MAX_SEARCH_RESULTS)
+                        sorted_results.pop_min();
+
+                    sorted_results.insert(match_result.score, path);
+                }
+            }
+
+            Vector<NonnullRefPtr<Result>> results;
+            results.ensure_capacity(sorted_results.size());
+            while (!sorted_results.is_empty()) {
+                auto score = sorted_results.peek_min_key();
+                auto path = sorted_results.pop_min();
+                results.append(adopt_ref(*new FileResult(path, score)));
```

Comment:
I don't think that you will ever need to write a new in this codebase, use `make_ref_counted` instead.

Reply:
I've changed this and updated all remaining usages of `adopt_ref()` in this file, so that we're being consistent.